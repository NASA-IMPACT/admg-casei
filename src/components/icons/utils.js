/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from "react"

import {
  AtmosphericCompositionIcon,
  // AtmosphericDynamicsIcon,
  CarbonCycleEcosystemsIcon,
  ClimateVariabilityChangeIcon,
  EarthSurfaceInteriorIcon,
  GlobalWaterEnergyCycleIcon,
  WeatherIcon,
  ExclamationIcon,
} from "../icons"

const icons = {
  "df6847ab-9443-4fdb-a990-d5bf1daca3b7": ({ color, size }) => (
    <AtmosphericCompositionIcon color={color} size={size} />
  ),
  "f1bd646f-876f-4e9f-896e-f7d4571ecc00": ({ color, size }) => (
    <CarbonCycleEcosystemsIcon color={color} size={size} />
  ),
  "cd62c68f-a9f7-456a-ae15-53ef79ed5118": ({ color, size }) => (
    <ClimateVariabilityChangeIcon color={color} size={size} />
  ),
  "2231a3f1-3430-4f89-86a2-94938aa000a5": ({ color, size }) => (
    <EarthSurfaceInteriorIcon color={color} size={size} />
  ),
  "fd5c0baa-d6a4-45d7-b201-0eb2a5505dc7": ({ color, size }) => (
    <GlobalWaterEnergyCycleIcon color={color} size={size} />
  ),
  "6d719465-f32c-4808-9efb-da5ad2bae135": ({ color, size }) => (
    <WeatherIcon color={color} size={size} />
  ),
}

export function getFocusIcon(id) {
  return Object.keys(icons).includes(id)
    ? icons[id]
    : ({ color, size }) => <ExclamationIcon color={color} size={size} />
}

export const sizes = {
  text: { width: "16", height: "16" },
  tiny: { width: "30", height: "30" },
  small: { width: "60", height: "60" },
  large: { width: "120", height: "120" },
  huge: { width: "360", height: "360" },
}
