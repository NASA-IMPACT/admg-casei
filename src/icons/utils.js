/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from "react"

import {
  AtmosphericCompositionIcon,
  // AtmosphericDynamicsIcon,
  CarbonCycleEcosystemsIcon,
  ClimateVariabilityChangeIcon,
  EarthSurfaceInteriorIcon,
  GlobalWaterEnergyCycleIcon,
  WeatherIcon,
  ExclamationIcon,
} from "./"

const icons = {
  "Atmospheric Composition": ({ color, size }) => (
    <AtmosphericCompositionIcon color={color} size={size} />
  ),
  "Carbon Cycle & Ecosystems": ({ color, size }) => (
    <CarbonCycleEcosystemsIcon color={color} size={size} />
  ),
  "Climate Variability & Change": ({ color, size }) => (
    <ClimateVariabilityChangeIcon color={color} size={size} />
  ),
  "Earth Surface & Interior": ({ color, size }) => (
    <EarthSurfaceInteriorIcon color={color} size={size} />
  ),
  "Global Water & Energy Cycle": ({ color, size }) => (
    <GlobalWaterEnergyCycleIcon color={color} size={size} />
  ),
  Weather: ({ color, size }) => <WeatherIcon color={color} size={size} />,
}

export function getFocusIcon(id) {
  return Object.keys(icons).includes(id)
    ? icons[id]
    : ({ color, size }) => <ExclamationIcon color={color} size={size} />
}

export const sizes = {
  text: { width: "16", height: "16" },
  tiny: { width: "30", height: "30" },
  small: { width: "60", height: "60" },
  large: { width: "120", height: "120" },
  huge: { width: "360", height: "360" },
}
