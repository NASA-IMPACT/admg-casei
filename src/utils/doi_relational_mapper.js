import { minBy } from "lodash"

export function doiRelationalMapper(doiList) {
  return doiList.map(doi => {
    const startdate = minBy(
      doi.campaigns,
      campaign => campaign.start_date
    )?.start_date
    const enddate = minBy(
      doi.campaigns,
      campaign => campaign.end_date
    )?.end_date
    return { enddate: enddate, startdate: startdate, ...doi }
  })
}
