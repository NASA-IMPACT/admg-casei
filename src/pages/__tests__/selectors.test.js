import React from "react"
import renderer from "react-test-renderer"
import * as Gatsby from "gatsby"

import Explore from "../explore"
import { selector } from "../../utils/filter-utils"
import {
  query,
  location,
  site,
  images,
  staticQueries,
} from "../../../test/__fixtures__"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")
useStaticQuery.mockImplementation(() => ({
  site,
  images,
  ...staticQueries.data,
}))

describe("Explore page", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Explore data={query.data} location={location} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("selector", () => {
  it("has two selector funtions", () => {
    const functions = selector()
    expect(Object.keys(functions)).toHaveLength(2)
  })
})
