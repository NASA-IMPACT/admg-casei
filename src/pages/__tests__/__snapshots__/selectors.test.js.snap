// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaigns page renders correctly 1`] = `
.c1:hover {
  opacity: 0.64;
}

.c1:active {
  -webkit-transform: translate(0,1px);
  -ms-transform: translate(0,1px);
  transform: translate(0,1px);
}

.c3 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.5);
  z-index: 9999;
  opacity: 0;
  -webkit-transition: opacity 400ms ease-in;
  transition: opacity 400ms ease-in;
}

.c4 {
  width: 400px;
  position: relative;
  padding: 32px;
  border-radius: 3px;
  background: #303641;
  pointer-events: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.5rem 0 2rem 0;
  min-height: 2rem;
}

.c2 div:only-child {
  margin-left: auto;
}

<div
  className="c0"
  id="top"
>
  <header>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
          "margin": "0 auto",
          "maxWidth": "1280px",
          "padding": "2rem 5rem",
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": "1.5rem",
            "lineHeight": "1.5rem",
            "margin": 0,
            "zIndex": 100,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "columnGap": "1rem",
              "display": "grid",
              "gridTemplateColumns": "3rem auto",
              "gridTemplateRows": "1.5rem 1.5rem",
              "textDecoration": "none",
            }
          }
        >
          <div
            style={
              Object {
                "gridArea": "1 / 1 / 3 / 2",
              }
            }
          >
            <img
              alt="NASA | CASEI logo"
              fluid={
                Object {
                  "aspectRatio": 1,
                  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsSAAALEgHS3X78AAACjElEQVQ4y2WVWW/MURjGZ7TWTi0htmmj2thFXbQuVAixlbQkgxJBkdAQtdXWZSpVVUtpIqSJihAJiStbQiTc+AJ8BnHjO8z4PfKc5Pj3n/zmPetz3nPe95xJFYvFlNCHHZfyR7kcGqG9UChcx96AC7AHFoZx9KVT8ZcUw6bhFNyHQ0xYDhWU58EqaIE+yEO155QGnaTYXHgITa4vgp0wHapiR1hkDW3dY5yLxGbCa3jD4JzbOuALXIU/cNweLolENsIxmBErlyIyDDWua7sdLkvkNGShC34z7hZ2Asgehss6plhQjbsS7reFLSFwVsEJ23KQvsFW9w9SXh0mToSPOj86SizeBEe0cmKBc7CUcUPYKbAMXipw2B3/FuBnPQ0jWK064PPYbGrpO6BFLTpI/QUoqrvhO9S7rxY6VWhXxY2KaDaKpDz94b4W6vIwi32AzckjeAbzPaZPk/qdLoreXXgLSolp8t4Dj3q7Tc7BUWUHKD9/wQaPy+vnJsyGtY7aU90ElzXxiXdxBc5oLE7sx553pEeoZ2LBizSscEMNjI+S+r0T/XHkbT2shGvMW+yA9rivP+XVtrhTHjR7a1WwHT7APW1RWRBFXZHtVtLTrmhP1fEFr3o9qNfp0uPcqsC2+RjKUonPZ/sVFsAmZUjo0HlVwza4BCctHrZZbtugI9KjAZMduNswi4XvYOcEQQXhkcvh8FuhLvJGWTDkre6DYXjl43inCxEGlkQ5OODyXiUxrHO9TGKJ16bBu/ikbPjvPY1Ec/ZC13ESVEaPbWf0MmUQVDrpQTmRfFuTonXKLR94paMr8eeOfAY+w0/aW6P56TGqQdRbanRadFEeBeVkHqv7flBBCULhb0BafwG7wPZR6IGaZQAAAABJRU5ErkJggg==",
                  "originalImg": "/static/c98d88bae1ac1d803f877ce6657b89ab/73f23/logo.png",
                  "originalName": "logo.png",
                  "presentationHeight": 62,
                  "presentationWidth": 62,
                  "sizes": "(max-width: 62px) 100vw, 62px",
                  "src": "/static/c98d88bae1ac1d803f877ce6657b89ab/73f23/logo.png",
                  "srcSet": "/static/c98d88bae1ac1d803f877ce6657b89ab/73f23/logo.png 62w",
                  "srcSetWebp": "/static/c98d88bae1ac1d803f877ce6657b89ab/766b3/logo.webp 62w",
                  "srcWebp": "/static/c98d88bae1ac1d803f877ce6657b89ab/766b3/logo.webp",
                  "tracedSVG": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='400'%20height='400'%20viewBox='0%200%20400%20400'%20preserveAspectRatio='none'%3e%3cpath%20d='M174%201l-2%202%2029%201%2029-1-2-2c1-1-8-1-27-1s-28%200-27%201m-14%2012c-1%202%201%203%2011%203%208%200%2011-1%2012-2%203-2%200-3-12-3-8%200-10%200-11%202m52-1c-5%202%201%203%2013%204%2012%200%2015-1%2014-3s-23-3-27-1m-60%20105v13l3-2c3-4%203-17%200-22l-2-3-1%2014m-71%2017l1%203c2%201%2036%202%2051%201l1-2c0-2-1-2-4-2H81m-15%2013H55c-8%200-10%201-10%202%200%202%201%202%2014%202l14-1c0-2-6-5-7-3m-54%2014v27c2%208%205-14%204-24-1-4-3-5-4-3m373%202c-1%204%201%2020%202%2022%202%202%203-3%203-15%200-8-1-10-2-10l-3%203M3%20172l-2%202c-1-1-1%207-1%2027%200%2018%200%2027%201%2026l2%202%201-28a842%20842%200%2000-1-29m394%2028c0%2027%200%2031%202%2028s2-52%200-55-2%201-2%2027M89%20179H73c-14%200-15%200-15%202%200%203%201%203%205%202h20c4-1%206-2%207-1%202%201%204%200%204-2s-4-3-5-1m298%2035l-1%2014c0%2012%200%2014%202%2013s2-3%202-13c0-14-1-17-3-14m-375%204c-2%205-1%2022%201%2023s3-2%203-12c0-7-1-10-2-12-2-1-2-1-2%201m48%201l7%201h18c3%201%204%201%207-1l2-2H76c-17%200-17%200-16%202m101%20167l-1%202c1%203%2033%203%2035%200%201-1%200-1-2-1-6-2-30-2-32-1m56%200c-2%201-3%202-2%203%203%202%2025%201%2026-1v-2h-24m-44%2013c3%202%2052%202%2055%200s-1-2-28-2c-26%200-30%200-27%202'%20fill='%23d3d3d3'%20fill-rule='evenodd'/%3e%3c/svg%3e",
                }
              }
              src="test-image.png"
            />
          </div>
          NASA | CASEI
          <p>
            Catalog of Archived Suborbital Earth Science Investigations
          </p>
        </a>
      </div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <nav
          style={
            Object {
              "zIndex": 100,
            }
          }
        >
          <ul
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "listStyle": "none",
                "margin": 0,
              }
            }
          >
            <li
              style={
                Object {
                  "margin": "0 1rem 0 0",
                }
              }
            >
              <a
                href="/explore/campaigns"
              >
                Explore
              </a>
            </li>
            <li
              style={
                Object {
                  "margin": "0 1rem 0 0",
                }
              }
            >
              <a
                href="/glossary"
              >
                Glossary
              </a>
            </li>
            <li
              style={
                Object {
                  "margin": "0 1rem 0 0",
                }
              }
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              style={
                Object {
                  "margin": "0 1rem 0 0",
                }
              }
            >
              <a
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <section
      data-cy="main-campaigns-section"
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": "1280px",
          "padding": "0 5rem",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-around",
            "marginBottom": "2rem",
          }
        }
      >
        <ul
          data-cy="tabbar"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "listStyle": "none",
              "margin": 0,
            }
          }
        >
          <li
            style={
              Object {
                "margin": "0 1rem 0 0",
              }
            }
          >
            <a
              href="/explore/campaigns"
            >
              Campaigns
            </a>
          </li>
          <li
            style={
              Object {
                "margin": "0 1rem 0 0",
              }
            }
          >
            <a
              href="/explore/platforms"
            >
              Platforms
            </a>
          </li>
          <li
            style={
              Object {
                "margin": "0 1rem 0 0",
              }
            }
          >
            <a
              href="/explore/instruments"
            >
              Instruments
            </a>
          </li>
        </ul>
      </div>
      <form
        data-cy="explore-tools"
        onSubmit={[Function]}
        style={
          Object {
            "display": "flex",
            "margin": "2rem 0",
          }
        }
      >
        <span
          id="filter-select"
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "whiteSpace": "nowrap",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          filter results by sub-categories
        </span>
        <div
          data-cy="filter-select"
          data-reach-listbox-input=""
          data-state="closed"
          data-value=""
          id="listbox-input--1"
        >
          <span
            aria-haspopup="listbox"
            aria-labelledby="filter-select button--listbox-input--1"
            data-reach-listbox-button=""
            id="button--listbox-input--1"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            style={
              Object {
                "WebkitAppearance": "none",
                "background": "transparent",
                "border": "1px solid hsla(0,0%,100%,0.9)",
                "borderRadius": "0.2rem 0 0 0.2rem",
                "color": "hsla(0,0%,100%,0.9)",
                "cursor": "pointer",
                "flexGrow": 0,
                "height": "2.5rem",
                "maxWidth": "5rem",
                "padding": "0.5rem",
              }
            }
            tabIndex={0}
          >
            Filter
            <span
              aria-hidden={true}
              data-reach-listbox-arrow=""
            >
              ▼
            </span>
          </span>
          <span />
        </div>
        <input
          data-reach-listbox-hidden-input=""
          disabled={false}
          name="filter"
          readOnly={true}
          tabIndex={-1}
          type="hidden"
          value=""
        />
        <div
          style={
            Object {
              "border": "1px solid hsla(0,0%,100%,0.9)",
              "borderLeft": 0,
              "borderRadius": "0",
              "borderRight": 0,
              "display": "flex",
              "flexGrow": 1,
              "padding": "0.25rem",
            }
          }
        >
          <input
            aria-label="Enter search text"
            autoComplete="off"
            data-cy="explore-input"
            name="search"
            placeholder="Enter text..."
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "color": "hsla(0,0%,100%,0.9)",
                "flexGrow": 1,
              }
            }
            type="text"
          />
          <button
            data-cy="submit"
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "color": "hsla(0,0%,100%,0.9)",
                "flexGrow": 0,
                "verticalAlign": "middle",
              }
            }
            type="submit"
          >
            <span
              aria-label="Magnifying glass icon"
              role="img"
            >
              <svg
                height="16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="16"
                  width="16"
                />
                <path
                  d="M15.708,13.587l-3.675-3.675C12.646,8.92,13,7.751,13,6.5C13,2.91,10.09,0,6.5,0S0,2.91,0,6.5S2.91,13,6.5,13 c1.251,0,2.42-0.354,3.412-0.967l3.675,3.675c0.389,0.389,1.025,0.389,1.414,0l0.707-0.707 C16.097,14.612,16.097,13.976,15.708,13.587z M3.318,9.682C2.468,8.832,2,7.702,2,6.5s0.468-2.332,1.318-3.182S5.298,2,6.5,2 s2.332,0.468,3.182,1.318C10.532,4.168,11,5.298,11,6.5s-0.468,2.332-1.318,3.182C8.832,10.532,7.702,11,6.5,11 S4.168,10.532,3.318,9.682z"
                  fill="hsla(0,0%,100%,0.9)"
                />
              </svg>
            </span>
          </button>
        </div>
        <span
          id="sort-order"
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "whiteSpace": "nowrap",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          Select sort order
        </span>
        <div
          data-cy="sort-select"
          data-reach-listbox-input=""
          data-state="closed"
          data-value="most recent"
          id="listbox-input--2"
        >
          <span
            aria-haspopup="listbox"
            aria-labelledby="sort order button--listbox-input--2"
            data-reach-listbox-button=""
            id="button--listbox-input--2"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            style={
              Object {
                "WebkitAppearance": "none",
                "background": "transparent",
                "border": "1px solid hsla(0,0%,100%,0.9)",
                "borderRadius": "0 0.2rem 0.2rem 0",
                "color": "hsla(0,0%,100%,0.9)",
                "cursor": "pointer",
                "height": "2.5rem",
                "padding": "0.5rem",
              }
            }
            tabIndex={0}
          >
            MOST RECENT
            <span
              aria-hidden={true}
              data-reach-listbox-arrow=""
            >
              ▼
            </span>
          </span>
          <span />
        </div>
      </form>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexWrap": "wrap",
            "margin": "2rem 0",
          }
        }
      >
        Active filters:
        <div
          data-cy="filter-chip"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#303641",
              "borderRadius": "0.2rem",
              "color": "hsla(0,0%,100%,0.9)",
              "display": "flex",
              "margin": "0.25rem 0.5rem",
              "padding": "0 0.5rem",
            }
          }
        >
          <small
            title=""
          >
            geophysical: Upper Atms Chem
          </small>
          <button
            className="c1"
            data-cy="remove-filter"
            onClick={[Function]}
            style={
              Object {
                "background": "none",
                "border": "none",
                "color": "hsla(0,0%,100%,0.9)",
                "cursor": "pointer",
                "flexGrow": 0,
                "marginLeft": "0.5rem",
                "verticalAlign": "middle",
              }
            }
            type="button"
          >
            <span
              aria-label="remove-filter-icon"
              role="img"
            >
              <svg
                height="12px"
                viewBox="0 0 16 16"
                width="12px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  fill="hsla(0,0%,100%,0.9)"
                  points="13.707,3.707 12.293,2.293 8,6.586 3.707,2.293 2.293,3.707 6.586,8 2.293,12.293 3.707,13.707 8,9.414  12.293,13.707 13.707,12.293 9.414,8"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        data-cy="mapboxgl-map"
        style={
          Object {
            "height": 500,
          }
        }
      />
      <section>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <h1
            data-cy="item-count"
            style={
              Object {
                "fontSize": "small",
                "whiteSpace": "nowrap",
              }
            }
          >
            Showing
             
            3 of 
            10
             
            campaigns
          </h1>
        </div>
        <div
          style={
            Object {
              "display": "grid",
              "gap": "1rem",
              "gridTemplateColumns": "repeat(auto-fill, minmax(270px, 1fr))",
            }
          }
        >
          <a
            href="/campaign/03b728ae-2235-488c-80ee-570dcf602f3b"
          >
            <div
              data-cy="explore-card"
              style={
                Object {
                  "backgroundColor": "#303641",
                  "boxShadow": "rgba(68, 63, 63, 0.08) 0px -1px 1px 0px, rgba(68, 63, 63, 0.08) 0px 2px 6px 0px",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "space-between",
                  "padding": "1rem",
                }
              }
            >
              <div
                style={
                  Object {
                    "marginBottom": "2rem",
                  }
                }
              >
                <div
                  className="c2"
                >
                  <img
                    height="85"
                    src="test-file-stub"
                  />
                </div>
                <big
                  data-cy="shortname"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "marginTop": "0.5rem",
                    }
                  }
                >
                  CARVE
                </big>
                <p
                  data-cy="longname"
                >
                  Carbon in Arctic Reservoirs Vulnerability Experiment
                </p>
                <div>
                  <small
                    data-cy="daterange"
                  >
                    2012—2015
                  </small>
                </div>
                <div>
                  <small
                    data-cy="region"
                  >
                    Alaska
                  </small>
                </div>
              </div>
              <div>
                <small
                  data-cy="count1"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <strong>
                    4
                  </strong>
                   
                  Deployment
                  s
                </small>
                <small
                  data-cy="count2"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                   · 
                  <strong>
                    0
                  </strong>
                   
                  Data Product
                  s
                </small>
              </div>
            </div>
          </a>
          <a
            href="/campaign/6d8b649b-e0b9-4adf-94bb-f5defc7ba930"
          >
            <div
              data-cy="explore-card"
              style={
                Object {
                  "backgroundColor": "#303641",
                  "boxShadow": "rgba(68, 63, 63, 0.08) 0px -1px 1px 0px, rgba(68, 63, 63, 0.08) 0px 2px 6px 0px",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "space-between",
                  "padding": "1rem",
                }
              }
            >
              <div
                style={
                  Object {
                    "marginBottom": "2rem",
                  }
                }
              >
                <div
                  className="c2"
                >
                  <img
                    height="85"
                    src="test-file-stub"
                  />
                </div>
                <big
                  data-cy="shortname"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "marginTop": "0.5rem",
                    }
                  }
                >
                  DC3
                </big>
                <p
                  data-cy="longname"
                >
                  Deep Convective Clouds & Chemistry
                </p>
                <div>
                  <small
                    data-cy="daterange"
                  >
                    2012
                  </small>
                </div>
                <div>
                  <small
                    data-cy="region"
                  >
                    Northeastern Colorado, West Texas to Central Oklahoma, Northern Alabama
                  </small>
                </div>
              </div>
              <div>
                <small
                  data-cy="count1"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <strong>
                    1
                  </strong>
                   
                  Deployment
                </small>
                <small
                  data-cy="count2"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                   · 
                  <strong />
                   
                  Data Product
                  s
                </small>
              </div>
            </div>
          </a>
          <a
            href="/campaign/e48c2d79-822b-4c9e-9f72-2776d5d1eac6"
          >
            <div
              data-cy="explore-card"
              style={
                Object {
                  "backgroundColor": "#303641",
                  "boxShadow": "rgba(68, 63, 63, 0.08) 0px -1px 1px 0px, rgba(68, 63, 63, 0.08) 0px 2px 6px 0px",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "space-between",
                  "padding": "1rem",
                }
              }
            >
              <div
                style={
                  Object {
                    "marginBottom": "2rem",
                  }
                }
              >
                <div
                  className="c2"
                >
                  <img
                    height="85"
                    src="test-file-stub"
                  />
                </div>
                <big
                  data-cy="shortname"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "marginTop": "0.5rem",
                    }
                  }
                >
                  ARCTAS
                </big>
                <p
                  data-cy="longname"
                >
                  Arctic Research of the Composition of the Troposphere from Aircraft and Satellites
                </p>
                <div>
                  <small
                    data-cy="daterange"
                  >
                    2008
                  </small>
                </div>
                <div>
                  <small
                    data-cy="region"
                  >
                    Alaska, Western Canada, California
                  </small>
                </div>
              </div>
              <div>
                <small
                  data-cy="count1"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <strong>
                    3
                  </strong>
                   
                  Deployment
                  s
                </small>
                <small
                  data-cy="count2"
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                   · 
                  <strong>
                    0
                  </strong>
                   
                  Data Product
                  s
                </small>
              </div>
            </div>
          </a>
        </div>
      </section>
    </section>
  </main>
  <footer
    style={
      Object {
        "backgroundColor": "#262A31",
        "marginTop": "5rem",
      }
    }
  >
    <div
      style={
        Object {
          "display": "grid",
          "flexShrink": 0,
          "gap": "1rem",
          "gridTemplateColumns": "repeat(12, 1fr)",
          "margin": "5rem auto",
          "maxWidth": "1280px",
          "padding": "2rem 5rem",
        }
      }
    >
      <div
        style={
          Object {
            "gridColumn": "1 / span 3",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": "30px",
              "fontWeight": "900",
              "lineHeight": "2.5rem",
              "marginBottom": "0.5rem",
              "textTransform": "uppercase",
            }
          }
        >
          NASA | CASEI
        </div>
        <p>
          NASA | CASEI
           is a comprehensive inventory containing information about all airborne and field campaigns as well as aircrafts, instruments, and data products.
        </p>
      </div>
      <div
        style={
          Object {
            "gridColumn": "5 / span 2",
          }
        }
      >
        <p
          style={
            Object {
              "fontWeight": "bold",
              "lineHeight": "3rem",
              "margin": "0 0 1.5rem 0",
            }
          }
        >
          Explore
        </p>
        <ul
          style={
            Object {
              "listStyle": "none",
              "margin": 0,
            }
          }
        >
          <li>
            <a
              href="/explore/campaigns"
            >
              Campaigns
            </a>
          </li>
          <li>
            <a
              href="/explore/platforms"
            >
              Platforms
            </a>
          </li>
          <li>
            <a
              href="/explore/instruments"
            >
              Instruments
            </a>
          </li>
        </ul>
      </div>
      <div
        style={
          Object {
            "gridColumn": "7 / span 2",
          }
        }
      >
        <p
          style={
            Object {
              "fontWeight": "bold",
              "lineHeight": "3rem",
              "margin": "0 0 1.5rem 0",
            }
          }
        >
          Resources
        </p>
        <ul
          style={
            Object {
              "listStyle": "none",
              "margin": 0,
            }
          }
        >
          <li>
            <a
              href="/glossary/"
            >
              Glossary
            </a>
          </li>
          <li>
            <button
              className="c1"
              onClick={[Function]}
              style={
                Object {
                  "background": "none",
                  "backgroundColor": "#294060",
                  "border": 0,
                  "borderRadius": "0.2rem",
                  "color": "white",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontWeight": "bold",
                  "minWidth": "2rem",
                  "padding": "0.25rem 0.75rem",
                  "textAlign": "center",
                  "textShadow": "none",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                }
              }
            >
              Maintenance login
            </button>
            <div
              className="c3"
              onClick={[Function]}
            >
              <div
                className="c4"
              >
                <form
                  data-cy="login-form"
                  onSubmit={[Function]}
                >
                  <h3>
                    Log In
                  </h3>
                  <label
                    htmlFor="username"
                  >
                    Username
                  </label>
                  <input
                    name="username"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    autoComplete="on"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                  <input
                    type="submit"
                  />
                </form>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        style={
          Object {
            "gridColumn": "9 / span 2",
          }
        }
      >
        <p
          style={
            Object {
              "fontWeight": "bold",
              "lineHeight": "3rem",
              "margin": "0 0 1.5rem 0",
            }
          }
        >
          Quick Links
        </p>
        <ul
          style={
            Object {
              "listStyle": "none",
              "margin": 0,
            }
          }
        >
          <li>
            <a
              href="/about/"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact/"
            >
              Contact us
            </a>
          </li>
        </ul>
      </div>
      <div
        style={
          Object {
            "gridColumn": "11 / span 2",
          }
        }
      >
        <p
          style={
            Object {
              "fontWeight": "bold",
              "lineHeight": "3rem",
              "margin": "0 0 1.5rem 0",
            }
          }
        >
          Organizations
        </p>
        <ul
          style={
            Object {
              "listStyle": "none",
              "margin": 0,
            }
          }
        >
          <li>
            <a
              data-cy="admg-website-link"
              href="https://earthdata.nasa.gov/esds/impact/admg"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "#aac9ff",
                }
              }
              target="_blank"
            >
              ADMG Website
            </a>
          </li>
          <li>
            <a
              data-cy="impact-website-link"
              href="https://earthdata.nasa.gov/esds/impact"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "#aac9ff",
                }
              }
              target="_blank"
            >
              Impact Website
            </a>
          </li>
        </ul>
      </div>
      <hr
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "gridColumn": "1 / span 12",
          }
        }
      />
      <div
        style={
          Object {
            "display": "flex",
            "gridColumn": "1 / span 12",
            "justifyContent": "space-between",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "fontWeight": "bold",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundImage": "url('https://www.nasa.gov/sites/default/files/thumbnails/image/nasa-logo-web-rgb.png')",
                "backgroundPosition": "center",
                "backgroundSize": "cover",
                "height": 100,
                "width": 100,
              }
            }
          />
          <div>
            <div>
              National Aeronautics and Space Administration
            </div>
            <div>
              NASA Official:
               
              <span
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Rahul Ramachandran
              </span>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "alignItems": "center",
              "color": "#81AEFB",
              "display": "flex",
              "fontWeight": "bold",
              "textTransform": "uppercase",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      © 
      2020
      , Built by
       
      <a
        data-cy="devseed-website-link"
        href="https://www.developmentseed.org"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#aac9ff",
          }
        }
        target="_blank"
      >
        Development Seed
      </a>
      .
    </div>
  </footer>
</div>
`;
